src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -31,7 +32,35 @@ public List<BookDto> findAll() {
 
     @Override
     public BookDto getBookById(Long id) {
-        Book bookById = bookRepository.getBookById(id);
-        return bookMapper.toDto(bookById);
+        return bookRepository.findById(id).stream()
+                .map(bookMapper::toDto)
+                .findFirst()
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "can`t find book with id: " + id));
+    }
+
+    @Override
+    public Book updateBookById(Long id, BookDto updatedBookDto) {
```

Comment:
so here, I think must be the same 
```suggestion
    public BookDto updateBookById(Long id, CreateBookRequestDto requestDto) {
```